### PR TITLE
Fixes/textbox caret index lost on focus lost

### PR DESCRIPTION
**Pull request by danwalmsley**
## What does the pull request do?

When a textbox has caret index and on windows alt-tab is pressed or another way textbox looses focus...
the caret index is reset to 0.

## What is the current behavior?

Hard coded reset to 0, but this is not correct way to reset selection.

## What is the updated/expected behavior with this PR?

reset selection to caret index instead.

## Checklist

- [x] Added unit tests (if possible)?
- [ ] Added XML documentation to any related classes?
- [ ] Consider submitting a PR to https://github.com/AvaloniaUI/Avaloniaui.net with user documentation

### CODE REVIEW

File: src/Avalonia.Controls/TextBox.cs [hunk 1]
```diff
@@ -407,14 +407,18 @@ protected override void OnGotFocus(GotFocusEventArgs e)
             _presenter?.ShowCaret();
         }
 
+        private void ClearSelection()
```
Comment (pr8x) [inline src/Avalonia.Controls/TextBox.cs:410]: Can we actually make this public? I recently had need for something like this

Comment (danwalmsley) [reply src/Avalonia.Controls/TextBox.cs]: good idea... added comments and made public.

Comment (pr8x) [reply src/Avalonia.Controls/TextBox.cs]: Is setting SelectedText=null doing the same actually?

**Comment by grokys** [comment]
I think we need to stop sending `LostFocus` to controls when the window is deactivated - that's clearly wrong behavior. But in the meantime if this PR works, i'm ok with it.

**Pull request merged**